mission_control/navigator_alarm/nodes/event_listeners/odom_kill_listener.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python
+import rospy
+from std_msgs.msg import String
+from navigator_alarm import single_alarm
+from twisted.internet import reactor
+import time
+import nav_msgs.msg import Odometry
+
+
+class OdomKillListener(object):
+    '''
+    Will kill the boat when it stops hearing odometry messages or when there is a large
+    discontinuity in state estimation. Only meant to run on the boad as a safety measure.
+    '''
+    def __init__(self, timeout=0.5):
+        self.timeout = rospy.Duration(timeout)
+        self.last_time = rospy.Time.now()
+        self.last_position = None
+        self.max_jump = 1.0
+        self.odom_discontinuity = False
+        self.sub = rospy.Subscriber('/odom', String, self.got_network_msg, queue_size=1)
+        self.alarm_broadcaster, self.alarm = single_alarm('kill', severity=3,
+                                                          problem_description="Fatal state estimation error")
+        rospy.Timer(rospy.Duration(0.1), self.check)
+
+    def check(self, *args):
+        if self.need_kill() and self.last_msg != None:
+            self.alarm.raise_alarm()
+            rospy.logerr("STATE ESTIMATION LOSS: KILLING BOAT")
+        else:
+            self.alarm.clear_alarm()
```

Comment:
It's a super easy change - instead of making a ros service host, you just create a listener class and define a callback, then when an event happens with the alarm (raised or cleared), that callback will get passed the alarm then you can check `alarm.clear` (or something like that) and make decisions based on that. As a result, this would cause the odom-kill alarm to be cleared anytime you cleared kill (that might be seen as a bug since any kill clear would clear the odom-kill but I don't think that will be a problem (P.S. meta alarms were meant to help with this issue (but I haven't delved to deeply into that hole (oh god the parenthesis)))).